### ISSUE TITLE
Tuna cURL Timeout causing OBS to freeze

### ISSUE THREAD

**Issue by TylerP5503** [univrsal/tuna#180]
When using Tuna everything works fine with viewing the name of song and the cover picture and etc and connecting spotify to Tuna, but It keeps crashing my OBS quite quickly, whenever I look in my log I see that it repeats:

[tuna] cURL request timeout over.
15:27:59.872: [tuna] cURL failed while sending spotify command (HTTP error -1, cURL error 28: 'Timeout was reached'). Waiting 5 seconds before trying again

Ive put the timeout refresh for 5000ms like what was said online to do to see about fixing the issue but its still not working, it just freezes my OBS completely and I have to end the task

**Log**
https://obsproject.com/logs/BYCybmNGZtWYruLL

**Comment by univrsal** [comment]
If requests to spotify frequently exceed a five second response time I'd first try to find out why that's the case.

**Comment by TylerP5503** [comment]
> If requests to spotify frequently exceed a five second response time I'd first try to find out why that's the case.

I honestly have no idea what the cause is, ive tried fixing things in firewall but still no luck, idk even where to start to figure out whats wrong with it timing out, do you know the possible issues why?

**Comment by univrsal** [comment]
No, but my best guess is that it's a connection issue. Try pinging `api.spotify.com` and see how fast it responds. Only other guess I have is that Spotify is rate-limiting.

**Comment by TylerP5503** [comment]
Hey, Ive still tried to fix things with Tuna and OBS, I got it to display stuff differently with the web browser instead of spotify, I cant seem to find anyone with the same issue as me, I updated my drivers again and it seems to just be worse, Im not getting the cURL error anymore since im using the web browser but ive noticed that when tuna is running it just completely causes my obs to freeze almost immediately when doing nothing or when i leave obs and do something else or when i click on things, Is the latest Tuna 1.9.7 supposed to be working with the latest stable OBS Studio 30.0.2 or do I need to downgrade obs? Ive also disabled HAGS and updated everything, thank you for you help, much appreciated :)

**Comment by TylerP5503** [comment]
I downgraded to OBS Studio 30 and it seems to have fixed the issue so far, will update if anything changes

**Comment by univrsal** [comment]
Can you post the log of you using obs studio 30?

**Comment by TylerP5503** [comment]
Yea ofc I gotcha, [](https://obsproject.com/logs/OqNupsa8wyNfsHJq)

**Comment by TylerP5503** [comment]
It seems to be doing much better, but I have gotten 1 freeze but that took quite a while to actually happen unlike when i had obs 30.0.2

**Comment by univrsal** [comment]
I don't think the version should make a difference. I thought maybe the Qt version changed but both 30 and 30.02 use the same qt version

**Comment by TylerP5503** [comment]
Okay well maybe not fully working, it froze again while trying to make the browser Tuna widget smaller so not sure whats going on there but its still seems to be doing better. Hmmm interesting well im not sure what fixed it a bit but switching versions has seemed to help some, not sure what qt stuff is but maybe it could be something else thats going on within it but idk about it so i could be wrong

**Comment by TylerP5503** [comment]
I have brower sources that seem to be having issues in the log with streamelements, could that be a cause maybe

**Comment by univrsal** [comment]
I don't think so.

**Comment by TylerP5503** [comment]
hmmm dang, well hopefully its nothing too bad to fix, ill keep trying things to fix it

**Comment by TylerP5503** [comment]
Its starting to have more freezing but not as much, it seems to be doing it mostly when im being active in obs such as clicking on things and sizing preview screen stuff, its like it gets overloaded with tasks or something but it shouldnt be being overloaded since my cpu and gpu are doing great since i have an intel i9 and a rtx 4080 so im just a bit confused if theres like a setting im missing or something, I disabled HAGS so it shouldnt be so bad and also made it a high priority both in graphic settings and on obs settings

**Comment by TylerP5503** [comment]
It seems to be causing freezes when i use either spotify option (causes cURL Timeout) or web browser (just freezes), but i dont receive full information like song timepoint/ progress and artist info when using windows media control/ other options

**Comment by TylerP5503** [comment]
Its like using anything web wise causes issues

**Comment by TylerP5503** [comment]
Windows media control works now but not fully, it cant seem to find the progress of the song, just the cover and song name and artist, do you know of a fix for that perhaps

**Comment by TylerP5503** [comment]
When I switch media players between opera and spotify it gets the time for a sec but it doesnt continue receiving the progress so it just stays where the timestamp was when i selected which media player
